dotnet/src/SemanticKernel/Connectors/HuggingFace/TextCompletion/HuggingFaceTextCompletion.cs

```diff
@@ -0,0 +1,162 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net.Http;
+using System.Text.Json;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.SemanticKernel.AI;
+using Microsoft.SemanticKernel.AI.TextCompletion;
+using Microsoft.SemanticKernel.Diagnostics;
+
+namespace Microsoft.SemanticKernel.Connectors.HuggingFace.TextCompletion;
+
+/// <summary>
+/// HuggingFace text completion service.
+/// </summary>
+public sealed class HuggingFaceTextCompletion : ITextCompletion, IDisposable
+{
+    private const string HttpUserAgent = "Microsoft Semantic Kernel";
+    private const string HuggingFaceApiBaseUri = "https://api-inference.huggingface.co";
+    private const string CompletionRemoteApiEndpoint = "/models";
+    private const string CompletionLocalApiEndpoint = "/completions";
+
+    private readonly string _completionEndpoint;
+    private readonly string _model;
+    private readonly HttpClient _httpClient;
+    private readonly HttpClientHandler? _httpClientHandler;
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="HuggingFaceTextCompletion"/> class.
+    /// </summary>
+    /// <param name="baseUri">Base URI for service API call.</param>
+    /// <param name="model">Model to use for service API call.</param>
+    /// <param name="httpClientHandler">Instance of <see cref="HttpClientHandler"/> to setup specific scenarios.</param>
+    public HuggingFaceTextCompletion(Uri baseUri, string model, HttpClientHandler httpClientHandler)
+    {
+        Verify.NotNull(baseUri, "Base URI cannot be null.");
+        Verify.NotEmpty(model, "Model cannot be empty.");
+
+        this._model = model;
+
+        this._httpClient = new(httpClientHandler);
+
+        this._httpClient.BaseAddress = baseUri;
+        this._httpClient.DefaultRequestHeaders.Add("User-Agent", HttpUserAgent);
+
+        this._completionEndpoint = CompletionLocalApiEndpoint;
+    }
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="HuggingFaceTextCompletion"/> class.
+    /// Using default <see cref="HttpClientHandler"/> implementation.
+    /// </summary>
+    /// <param name="baseUri">Base URI for service API call.</param>
+    /// <param name="model">Model to use for service API call.</param>
+    public HuggingFaceTextCompletion(Uri baseUri, string model)
```

Comment:
I think this method could be confusing, e.g. one will pass "https://api-inference.huggingface.co/" and wonder how to pass the API key. The other method accepting only the API key is not as obvious, e.g. I would add a default URI in there. Something like:

```
public HuggingFaceTextCompletion(string apiKey, string model, HttpClientHandler httpClientHandler, string host = "https://api-inference.huggingface.co/")
```

Reply:
good point, updated.